Assets/Scripts/Hoop.cs

```diff
@@ -5,12 +5,60 @@
 public class Hoop : ABaseEntity
 {
     public int value = 1;
+    private int currentSpeed;
+    private Vector3 startPos;
+    private float r;
+    private float startHeight;
+    public int[] speeds = { 0, 5, 15, 30, 50 };
+    public int currentPattern = 0;
+    public Player player;
     public ScoreManager scoreManager;
+    public float vertMoveScale = 1.0f;
     void OnTriggerEnter(Collider other)
     {
         if (other.CompareTag("Basketball"))
         {
             scoreManager.Score += value;
         }
     }
+
+    public override void BaseStart()
+    {
+        Vector3 pos = new Vector3(player.transform.position.x, 0, player.transform.position.z);
+
+        startPos.Set(this.transform.position.x, 0, this.transform.position.z);
+        currentSpeed = speeds[0];
+        r = Vector3.Distance(startPos, pos);
+        startHeight = this.transform.position.y;
+    }
+    public override void BaseUpdate()
+    {
+        if (scoreManager.Difficulty <= speeds.Length && currentSpeed != speeds[scoreManager.Difficulty])
+        {
+            currentSpeed = speeds[scoreManager.Difficulty];
+            currentPattern++;
+            if (currentPattern == 3)
+            {
+                startPos.Set(this.transform.position.x, 0, this.transform.position.z);
+            }
+        }
+
+        switch (currentPattern)
+        {
+            case 4:
+            case 3:
+                Vector3 pos = new Vector3(this.transform.position.x, 0, this.transform.position.z);
+                float theta = (Vector3.SignedAngle(startPos, pos, Vector3.up));
+                this.transform.position = new Vector3(this.transform.position.x, startHeight + (Mathf.Sin(theta * Mathf.Deg2Rad) * vertMoveScale), this.transform.position.z);
+                goto case 2;

```

Comment:
I could be wrong but I think by leaving out the `break` statement it will automatically go to case 2 anyways

Reply:
So part of the reason I used a switch case was that I thought it would do that too, but c#'s switch statements don't allow fall-throughs, so you have to end each case with either a break or goto case.
[https://stackoverflow.com/questions/174155/switch-statement-fallthrough-in-c](url)